### ISSUE TITLE
Poor rendering performance on pip install

### ISSUE THREAD

**Issue by waldie11** [mne-tools/mne-qt-browser#302]
### Description of the problem

Rendering performance on `pip install "mne[full-pyqt6]"` does not compare to the version provided on conda-forge.

### Steps to reproduce

1. install via
```Bash
pip install "mne[full-pyqt6]"
```

2. create script
```Python
import mne

sample_data_folder = mne.datasets.sample.data_path()
sample_data_raw_file = sample_data_folder / "MEG" / "sample" / "sample_audvis_raw.fif"
raw = mne.io.read_raw_fif(sample_data_raw_file, preload=True)

# ensure usage of qt backend
mne.viz.set_browser_backend("qt")

raw.plot(block=True)
```

3. run by
```Bash
QT_QPA_PLATFORM=xcb ipython --matplotlib=qt script_name.py
```

4. use scroll (by mouse wheel or along bars) in the qt browser window

### Expected results

smooth operation of scrolling along the axises.
This is true on the very same system for a conda-forge install of `mamba install mne`

### Actual results

The `pip install "mne[full-pyqt6]"` version suffers from poor rendering performance. Scrolling along the bars is painfully slow.

### Additional information

Overall goal is to use mne-qt-browser in a native wayland window. For this, firing off `QT_QPA_PLATFORM=wayland ipython --matplotlib=qt script_name.py` is coming up on the `pip install "mne[full-pyqt6]" version with identical bad performance, while the conda-forge version of mne pulls in qt5 which is not configured with wayland support.

I observe this behavior on both opensuse tumbleweed with kde plasma in wayland session, as well as ubuntu 22.04 and gnome3. Starting point for each flavor is a recent mambaforge environment with python 3.12.

**Comment by mscheltienne** [comment]
I'll have a look next week, but my office computer runs Ubuntu 24.04 with Wayland, and I never had performance issue with the pip-version of pyqt6 or with the qt browser which I use with pyqt6.

**Comment by drammock** [comment]
I can reproduce. Notably, it is not fixed by:

- disabling overview bar
- disabling projectors
- disabling antialiasing

I thought maybe it was the recent (Dec 6) release of `PyQt6-Qt6` version 6.8.1, but it turns out that wasn't even installed (the latest `PyQt6` version doesn't yet support 6.8.1). Upgrading `PyQt6-Qt6` to version 6.8.1 (isn't supposed to work and) didn't improve anything.

incidentally, I noticed that after disabling antialiasing, closing the settings menu, then re-opening settings, the antialiasing tickbox was once again checked.

**Comment by larsoner** [comment]
>  the conda-forge version of mne pulls in qt5 which is not configured with wayland support.

FWIW you can do something like
```
$ conda remove mne pyqt qt-main
$ conda install mne-base mne-qt-browser pyside6
```
or similar and it should allow you to test with PySide6, which should be the way of the future for conda at least (and are the official Python bindings of Qt, with a generally better license).

Can you test that? I think it's a more important target, and is at least another data point. I can also test next week

**Comment by larsoner** [comment]
> incidentally, I noticed that after disabling antialiasing, closing the settings menu, then re-opening settings, the antialiasing tickbox was once again checked.

Are you up to date @drammock ? This should have been fixed by https://github.com/mne-tools/mne-qt-browser/pull/271

**Comment by larsoner** [comment]
I tried to replicate the slowness on PySide6 and PyQt6 but both were fast for me with `QT_QPA_PLATFORM=wayland mne browse_raw ~/mne_data/MNE-sample-data/MEG/sample/sample_audvis_raw.fif`, and I made sure that `echo $XDG_SESSION_TYPE` was `wayland`. This was on Ubuntu 24.10 and a pretty new NVIDIA graphics card running proprietary drivers. I wonder if it's perhaps a more general driver/performance issue?

**Comment by waldie11** [comment]
> or similar and it should allow you to test with PySide6, which should be the way of the future for conda at least (and are the official Python bindings of Qt, with a generally better license).
> 
> Can you test that? I think it's a more important target, and is at least another data point. I can also test next week

No joy. PySide6, mne-base and mne-qt-browser all from conda-forge suffer from same issues as initially reported, including no native wayland support. 

> I tried to replicate the slowness on PySide6 and PyQt6 but both were fast for me with QT_QPA_PLATFORM=wayland 
> mne browse_raw ~/mne_data/MNE-sample-data/MEG/sample/sample_audvis_raw.fif,

If I do this with pyside6, I am facing:

qt.qpa.plugin: Could not find the Qt platform plugin "wayland" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

>  I wonder if it's perhaps a more general driver/performance issue?

This I am doubting, especially as it is working flawlessly with xcb/non-wayland qt5 version from conda-forge.
Also tests on Ubuntu 22.04 are done on two separate systems. All have the nvidia-open (non-nouveau) drivers though.

**Comment by waldie11** [comment]
> I wonder if it's perhaps a more general driver/performance issue?

Well - seems like it :relaxed:  I managed to get the pyside approach on native wayland up and running on Ubuntu after additionally installing `qt6-wayland` via apt in the base system.

Whatever setup my opensuse install is lacking, I will need to figure out myself I fear. Thanks for the hint!

**Comment by waldie11** [comment]
Tracked it down to missing updates/version match in the flatpak environment. Apparently I never tried execution of the small scripts from a pure mamba environment, so I must have always started from the flatpak VS Code shell.

So propperly providing OpenGl within flatpak apps after every big driver update is the way to go...
```
3. [✓] org.freedesktop.Platform.GL.default                      23.08                  u             flathub                 82,3 MB / 168,3 MB
 4. [✓] org.freedesktop.Platform.GL.default                     23.08-extra        u             flathub                 24,2 MB / 168,3 MB
 5. [✓] org.freedesktop.Platform.GL.default                     24.08                  u             flathub                156,3 MB / 156,3 MB
 6. [✓] org.freedesktop.Platform.GL.default                     24.08extra         u             flathub                 24,3 MB / 156,3 MB
 7. [✓] org.freedesktop.Platform.GL.nvidia-565-57-01     1.4                     u             flathub                339,3 MB / 340,2 MB
 8. [✓] org.freedesktop.Platform.Locale                           23.08                   u             flathub                100,1 KB / 371,7 MB
 9. [✓] org.freedesktop.Platform.Locale                           24.08                   u             flathub                 52,9 KB / 379,7 MB
10. [✓] org.freedesktop.Platform.VAAPI.Intel                  23.08                   u             flathub                 11,7 MB / 13,6 MB
11. [✓] org.freedesktop.Platform.VAAPI.Intel                  24.08                   u             flathub                 12,9 MB / 14,7 MB
12. [✓] org.freedesktop.Platform.ffmpeg-full                  23.08                   u             flathub                  5,7 MB / 9,2 MB
13. [✓] org.freedesktop.Platform.ffmpeg-full                  24.08                   u             flathub                  7,2 MB / 12,2 MB
14. [✓] org.freedesktop.Platform                                       23.08                   u             flathub                 11,7 MB / 230,8 MB
15. [✓] org.freedesktop.Platform                                       24.08                   u             flathub                 26,7 MB / 261,4 MB
```

@larsoner I am experiencing some worse performance on pure wayland still - should I open a propper issue for that? It is on a completely different level to the performance issue of misconfigured opengl rendering I ran into here though.

**Comment by drammock** [comment]
> > incidentally, I noticed that after disabling antialiasing, closing the settings menu, then re-opening settings, the antialiasing tickbox was once again checked.
> 
> Are you up to date @drammock ? This should have been fixed by #271

I was on 94fb5e8 when I tested this, so yeah, long after #271 got in.
